hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/metrics/DataNodePeerMetrics.java

```diff
@@ -58,11 +61,13 @@
    * for outlier detection. If the number of samples is below this then
    * outlier detection is skipped.
    */
-  @VisibleForTesting
-  static final long MIN_OUTLIER_DETECTION_SAMPLES = 1000;
+  private final long minOutlierDetectionSamples;
 
-  public DataNodePeerMetrics(final String name) {
+  public DataNodePeerMetrics(final String name, Configuration conf) {
```

Comment:
This one arguably breaks compatibility. But given that this class is
@InterfaceAudience.Private
@InterfaceStability.Unstable
I think this is acceptable.